internal/controllers/addoninstance_watcher.go

```diff
@@ -0,0 +1,65 @@
+package controllers
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	addonsv1alpha1 "github.com/openshift/addon-operator/apis/addons/v1alpha1"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+
+	"github.com/openshift/reference-addon/internal/addonsdk"
+)
+
+type AddonInstanceWatcher struct {
```

Comment:
Why is this a `*Watcher` when everything else in this package is a `*Reconciler`?

Reply:
No substantial reason, just wanted to differentiate it from something not directly associated the functional use-case of the addon (reference-addon) but yeah, I see your point, will shift it to Reconciler.